ministryofjustice/analytics-platform-control-panel-frontend
# Ask users to verify their email address on first login

## What

* Add email verification form and handler
* Add flash message middleware to notify user of successful update
* Update base model proxy to allow setting properties


## How to review

NOTE: Depends on https://github.com/ministryofjustice/analytics-platform-control-panel/pull/76

1. `docker-compose pull` to get the latest API image
2. `docker-compose up`
3. `docker-compose exec api python3 manage.py shell`
4. Make yourself a normal user:
    ```python
    from control_panel_api.models import User
    u = User.objects.get(username="your username")
    u.is_superuser = False
    u.save()
    ```
5. Browse to http://localhost:3000/ and login
6. See that you are redirected to a form to verify your email address
7. Submit the form with a changed email address and be redirected to the normal user home page
8. Logout and login again and see that you go straight to the home page
9. See that your email address has been updated:
    ```python
    u = User.objects.get(username="your username")
    assert u.email == "submitted email address"
    assert u.verified_email is True
    ```

See [Trello #453](https://trello.com/c/REq10r8F)
